### ISSUE TITLE
Ecoflow 1.2.0 After 2  Home Assistant Reboots Data are not Updated

### ISSUE THREAD

**Issue by leotzian** [tolwi/hassio-ecoflow-cloud#355]
Hi to all I know that this issue has been discussed some times in the past, but I want to make one question to all of you, Ecoflow public API that was added with the new version is better than application API, but the problem is that HA must restarted often because integrations are updated HA is updated, Is there a way for this problem to solve or is there any other solution, because it is difficult every some days to delete keys and generate new one's. If there is no way to find a solution it is better to return to application API. Thanks and sorry for opening this issue but this is a big issue.

**Comment by giovanne123** [comment]
I am interested how you added the devices, one group? Because it seems that a mqtt connection is done by group and if you have some groups the client ids per days can be reached and the reason for your issues, I still investigate this question:
https://github.com/tolwi/hassio-ecoflow-cloud/issues/351

Can you add a screenshot of your added devices to the integration? (all in one group or seprate as mine in linked issue.)

**Comment by leotzian** [comment]
Yes perhaps that's it, I didn't know how to add in one group, now I know and yes I have 6 groups as the number of the devices I have

**Comment by giovanne123** [comment]
I think delete and recreate would be the way to go... I can't confirm so far because it is a todo for me too.

**Comment by leotzian** [comment]
Is it certain that if all devices are in one group solves the problem? Had anyone tried this? In my case with 6 groups with one device in each, the problem of data not updating occur after 2 HA reboots.

**Comment by unze** [comment]
I have the same problem.
when i open the ecoflow app, i get data again for a minute or two.
i only have two devices in one group each

**Comment by giovanne123** [comment]
> I have the same problem. when i open the ecoflow app, i get data again for a minute or two. i only have two devices in one group each

To not mix everything up, we were talking about public API, you too? In my case no need for the App, issue related to client ids which may be recreated when HA needs restart after HA update, or similiar situations.

**Comment by unze** [comment]
Soory, my fault. Private API

**Comment by leotzian** [comment]
The best is @tolwi when have time reply if groups has to do anything with this issue and if is better to have all devices to one group.

**Comment by giovanne123** [comment]
Right, as already noticed here [#351 ](https://github.com/tolwi/hassio-ecoflow-cloud/issues/351) question if groups overall are necessary or can be removed because grouping can be done in HA itself.

**Comment by leotzian** [comment]
The question is simple if we have 2 cases, one 6 devices in a group, and the second the same 6 devices in 6 groups one device in every group are the client I'ds request in the second case would be six times more or the same? 
The person that can answer this question is the one that had developed this integration.

**Comment by andreasbuff** [comment]
I think I found a solution for the problem, that data are not updated?! See link: https://github.com/tolwi/hassio-ecoflow-cloud/issues/361#issuecomment-2406929316

**Comment by justnx** [comment]
All that has not helped in my case. It seems more related to their API. It seems it was down, since i was not able to access my key using the developer account. today service seems to have recovered and i deleted my api key and recreated it. with the new key the integration worked again.

**Comment by Ouka-miko** [comment]
https://github.com/tolwi/hassio-ecoflow-cloud/discussions/302#discussioncomment-10583408
https://github.com/tolwi/hassio-ecoflow-cloud/issues/289#issuecomment-2330998952
